packages/core/src/shaderlib/mobile_blinnphong_frag.glsl

```diff
@@ -7,13 +7,19 @@
 
     vec3 lightDiffuse = vec3( 0.0, 0.0, 0.0 );
     vec3 lightSpecular = vec3( 0.0, 0.0, 0.0 );
+    float shadowAttenuation = 1.0;
 
     #ifdef O3_DIRECT_LIGHT_COUNT
-    
-    DirectLight directionalLight;
+    shadowAttenuation = 1.0;
+#ifdef OASIS_CALCULATE_SHADOWS
+    #ifdef CASCADED_SHADOW_MAP
+        shadowAttenuation *= sampleShadowMap();
+    #endif
+#endif
 
+    DirectLight directionalLight;
     for( int i = 0; i < O3_DIRECT_LIGHT_COUNT; i++ ) {
-        directionalLight.color = u_directLightColor[i];
+        directionalLight.color = u_directLightColor[i] * shadowAttenuation;
```

Comment:
shadowAttenuations shouldn't apply every light, should just light which enable cast shadow